public/framework/catalog_cache/cache_loader.tsx

```diff
@@ -0,0 +1,268 @@
+    name: row.tableName,
+    columns: [],
+  }));
+
+  CatalogCacheManager.updateDatabase(dataSourceName, {
+    ...cachedDatabase,
+    tables: newTables,
+    lastUpdated: currentTime,
+    status: CachedDataSourceStatus.Updated,
+  });
+};
+
+export const updateAccelerationsToCache = (pollingResult: any) => {
+  const currentTime = new Date().toUTCString();
+
+  if (!pollingResult) {
+    CatalogCacheManager.saveAccelerationsCache({
+      version: '1.0',
+      accelerations: [],
+      lastUpdated: currentTime,
+      status: CachedDataSourceStatus.Failed,
+    });
+    return;
+  }
+
+  const combinedData = combineSchemaAndDatarows(pollingResult.schema, pollingResult.datarows);
+
+  const newAccelerations = combinedData.map((row: any) => ({
+    flintIndexName: row.flint_index_name,
+    type: row.kind === 'mv' ? 'materialized' : row.kind,
+    database: row.database,
+    table: row.table,
+    indexName: row.index_name,
+    autoRefresh: row.auto_refresh,
+    status: row.status,
+  }));
+
+  CatalogCacheManager.saveAccelerationsCache({
+    version: '1.0',
+    accelerations: newAccelerations,
+    lastUpdated: currentTime,
+    status: CachedDataSourceStatus.Updated,
+  });
+};
+
+export const updateToCache = (
+  pollResults: any,
+  loadCacheType: LoadCacheType,
+  dataSourceName: string,
+  databaseName?: string
+) => {
+  switch (loadCacheType) {
+    case 'databases':
+      updateDatabasesToCache(dataSourceName, pollResults);
+      break;
+    case 'tables':
+      updateTablesToCache(dataSourceName, databaseName!, pollResults);
+      break;
+    case 'accelerations':
+      updateAccelerationsToCache(pollResults);
+      break;
+    default:
+      break;
+  }
+};
+
+export const createLoadQuery = (
+  loadCacheType: LoadCacheType,
+  dataSourceName: string,
+  databaseName?: string
+) => {
+  let query;
+  switch (loadCacheType) {
+    case 'databases':
+      query = `SHOW SCHEMAS IN ${addBackticksIfNeeded(dataSourceName)}`;
+      break;
+    case 'tables':
+      query = `SHOW TABLES IN ${addBackticksIfNeeded(dataSourceName)}.${addBackticksIfNeeded(
+        databaseName!
+      )}`;
+      break;
+    case 'accelerations':
+      query = `SHOW FLINT INDEX in ${addBackticksIfNeeded(dataSourceName)}`;
+      break;
+    default:
+      query = '';
+      break;
+  }
+  return query;
+};
+
+export const useLoadToCache = (loadCacheType: LoadCacheType) => {
+  const sqlService = new SQLService(coreRefs.http!);
+  const [currentDataSourceName, setCurrentDataSourceName] = useState('');
+  const [currentDatabaseName, setCurrentDatabaseName] = useState<string | undefined>('');
+  const [loadStatus, setLoadStatus] = useState<DirectQueryLoadingStatus>(
+    DirectQueryLoadingStatus.SCHEDULED
+  );
+
+  const {
+    data: pollingResult,
+    loading: _pollingLoading,
+    error: pollingError,
+    startPolling,
+    stopPolling: stopLoading,
+  } = usePolling<any, any>((params) => {
```

Comment:
I know linting complains a lot about `any`: it is ok here? Better than `unknown`?

Reply:
I tried `unknown` earlier but that gives a TS error. `any` works right now cause the usePolling hook uses `any`  internally.